src/test/java/net/fornwall/jelf/ElfParserTest.java

```diff
@@ -0,0 +1,139 @@
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(
+                new byte[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 0});
+        ElfParser elfParser = new ElfParser(elfFile, byteArrayInputStream);
+
+        Assert.assertEquals(16909060, elfParser.readInt());
+    }
+
+    @PrepareForTest(ElfFile.class)
+    @Test
+    public void testReadLong() {
+        ElfFile elfFile = PowerMockito.mock(ElfFile.class);
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(
+                new byte[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 0});
+        ElfParser elfParser = new ElfParser(elfFile, byteArrayInputStream);
+
+        Assert.assertEquals(72623859790382856L, elfParser.readLong());
+    }
+
+    @PrepareForTest(ElfFile.class)
+    @Test
+    public void testReadIntOrLong() {
+        ElfFile elfFile = PowerMockito.mock(ElfFile.class);
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(
+                new byte[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 0});
+        ElfParser elfParser = new ElfParser(elfFile, byteArrayInputStream);
+
+        Assert.assertEquals(72623859790382856L, elfParser.readIntOrLong());
+    }
+
+    @PrepareForTest(ElfFile.class)
+    @Test
+    public void testUnsignedByte() {
+        ElfFile elfFile = PowerMockito.mock(ElfFile.class);
+        ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(
+                new byte[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 0});
+        ElfParser elfParser = new ElfParser(elfFile, byteArrayInputStream);
+
+        Assert.assertEquals(123, elfParser.unsignedByte(123));
+    }
+
+    @PrepareForTest(MemoizedObject.class)
+    @Test
+    public void testVirtualMemoryAddrToFileOffset() throws Exception {
+        ElfParser elfParser = (ElfParser)
+                Reflector.getInstance("net.fornwall.jelf.ElfParser");
+        ElfFile elfFile = (ElfFile)
+                Reflector.getInstance("net.fornwall.jelf.ElfFile");
+
+        ElfSegment elfSegment = (ElfSegment)
+                Reflector.getInstance("net.fornwall.jelf.ElfSegment");
+        Reflector.setField(elfSegment, "mem_size", 4_611_685_010_250_924_278L);
+        Reflector.setField(elfSegment, "virtual_address", 1_008_176_471_803L);
+        Reflector.setField(elfSegment,
+                "file_size", 9_223_371_028_678_312_157L);
+
+        MemoizedObject memoizedObj = PowerMockito.mock(MemoizedObject.class);
+        Method getValueMethod =
+                DTUMemberMatcher.method(MemoizedObject.class, "getValue");
+        PowerMockito.doReturn(elfSegment)
+                .when(memoizedObj, getValueMethod).withNoArguments();
+
+        MemoizedObject[] memoizedObjectArray = {memoizedObj};
+        Reflector.setField(elfFile, "programHeaders", memoizedObjectArray);
+        Reflector.setField(elfFile, "num_ph", (short) 1);
+        Reflector.setField(elfParser, "fsFile", null);
+        Reflector.setField(elfParser, "elfFile", elfFile);
```

Comment:
The `ElfParser` can be constructed using `new ElfParser(elfFile, null)`

Reply:
Done